aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt

```diff
@@ -98,26 +98,26 @@ class AztecMediaSpan(context: Context, drawable: Drawable?, var attributes: Attr
         val sb = StringBuilder()
         sb.append("<")
         sb.append(TAG)
+        sb.append(' ')
 
-        attributes?.let {
-            for (i in 0..attributes!!.length-1) {
-                sb.append(' ')
-                sb.append(attributes!!.getLocalName(i))
-                sb.append("=\"")
-                sb.append(attributes!!.getValue(i))
-                sb.append("\"")
-            }
-        }
+        attributes.removeAttribute("aztec_id")
 
+        sb.append(attributes)
         sb.append("/>")
+
         return sb.toString()
     }
 
     fun onClick(view: View) {
+        // add a unique ID that can identify this image span
```

Comment:
> Hmm, that sounds like something the client should do, not Aztec

I'd agree but if you can't identify an element you can't add attributes to it from the client :).

>  I guess it's an opportunistic setting but still, too "hackish". Also, the attribute name itself seems "magical", in the sense that it's hardcoded and not advertised (i.e. a public constant) anywhere.

Perhaps. Maybe it should a more general rule - maybe it should be added to every element, not just images. The "id property" name could then be exposed through some public constant in `AztecText` and the client would know that it's always there if necessary.

> Even if we keep the idea of performing this in onClick, the client can do that, if it wishes to, in its onMediaTappedListener?.mediaTapped() implementation.

As mentioned above, if you don't have an ID already, you won't be able to find the element to add it.